### ISSUE TITLE
inherit parent data attributes for tracking 

### ISSUE THREAD

**Issue by mmairs9** [nfl/react-metrics#33]
Is there any way to inherit higher level component or parent elements data-attributes to its children so I don't have to duplicate data attributes throughout the codebase?

I was able to create a [PR](https://github.com/nfl/react-metrics/pull/32) but it broke some of the tests. Basically instead of traversing through the next specified root element (MetricsElement) it will traverse to the root Dom element on document body. It will then add any parent data attributes to the existing track call. Lower level attributes will override their parent as it is closer to the source of the click tracking.

**Comment by taak77** [comment]
@mmairs9 thanks for the [PR](https://github.com/nfl/react-metrics/pull/32), i think there is a use case where it's convenient, but I want to only opt in this feature, so i made this [PR](https://github.com/nfl/react-metrics/pull/34) on top of your commit, added test and updated example. Basically the root element you want to start sharing metrics data attributes needs to be `MetricsElement`. Check out the [PR](https://github.com/nfl/react-metrics/pull/34), run the example(http://localhost:8080/react-router/metricsElement/index.html) and let me know if that works for you.

**Comment by mmairs9** [comment]
@taak77 this looks great, thanks for looking into this.  We have the need to define an area and sections within an area on a page so inheriting these data attributes instead of having to define them in each tracking element will be great! 

With the approach you taken could we make it so we can nest Metrics Elements within each other?This would make it so when I define a click event on an element with a section it is not only able to see the parent data attribute on that metric element for section but in a parent Metrics element as well which was defined for the area. Then on the track call it will add these section and area variables without having to define them on each child tracking element.

**Comment by taak77** [comment]
@mmairs9 not sure if i follow, but in this use case you will only need `MetricsElement` on your root where you want to start sharing the metrics data attribute as any click event in children will be captured by event delegation.

```
<MetricsElement
    element="div"
    data-metrics-page="page A"
>
    <div
        data-metrics-merge-pagedefaults="true"
        data-metrics-section="section 1"
        data-metrics-event-name="imageClick"
    >
        <a data-metrics-value="Image 1">
            <img src="http://placehold.it/200x150?text=Image+1" />
        </a>

        <a data-metrics-value="Image 2">
            <img src="http://placehold.it/200x150?text=Image+2" />
        </a>
    </div>

    <div
        data-metrics-section="section 2"
        data-metrics-event-name="listItemClick"
    >
        <ul>
            <li data-metrics-value="list item 1"><span>Item 1</span></li>
            <li data-metrics-value="list item 2"><span>Item 2</span></li>
            <li data-metrics-value="list item 3"><span>Item 3</span></li>
        </ul>
    </div>
</MetricsElement>
```

Does it cover what you mentioned?

**Comment by mmairs9** [comment]
@taak77 I just pulled your latest commit into our project and it works great, so you could ignore my last comment (thanks for the example). Just noticed one slight problem. I am surrounding my markup in a parent component with one Metrics element so i can share analytic data downstream to other children components with tracking elements. When I click anywhere that does not have a data metric-event-name defined I receive the following console error

Uncaught Error: Metrics 'track' method requires 'eventName' string as the first argument and object or promise as the second argument.

 Can we suppress this error for this use case?

**Comment by taak77** [comment]
@mmairs9 thanks for catching the error and just updated #34, can you try again? It's technically a breaking change now.

**Comment by mmairs9** [comment]
@taak77 Thanks! This works great now after your last commit.

**Comment by taak77** [comment]
@mmairs9 published as 2.3.0, please try out, also can you close your [PR](https://github.com/nfl/react-metrics/pull/32) now?

**Comment by mmairs9** [comment]
@taak77 closed the PR, thanks again!
